### DISCUSSION TITLE
Migrate data between apps

### DISCUSSION THREAD

**Discussion by hugopvl** [directus/directus#9009] [Archive]
Good afternoon, we have an application that is already obsolete, but it contains a lot of important information, in reality they are records, with images and attachments in pdf, we would like to migrate all this information to directus, but as it is a lot of information, it is unthinkable to do so manually.

the original information is in a sql DB, as well as the directus. all text type information is easily resolved with a query/update, but images and attachments are a problem.

can you guide me how best to do it?

Thank you.

**Comment by mikkel1156** [comment]
I think it depends how you want to store your data. If it's data that has relation to the records, you could have it in a blob/binary column (Directus has binary fields too). Otherwise your other option would be to upload it as assets as seen here https://docs.directus.io/reference/api/system/files/#upload-a-file.

**Comment by rijkvanzanten** [comment]
Besides the raw assets, do you have any / a lot of relational fields pointing to those assets? You could theoretically manually generate the directus_files records so you don't have to re-upload everything, however, that won't solve for the foreign keys in relationships that you might have to update 🤔

**Comment by joselcvarela** [comment]
I already experienced the same struggle almost in every project I have worked. 
You can say: "Just install Directus on top of your database".
But, things like, file associations, translations, associate proper interfaces, etc. Those are all tasks everyone need to do in order to use an existing database.

I don't know if this fits the 80/20 rule but I think this could improve our market share.
Like LDAP, this are features the other CMS don't usually have, but people and specially companies wish to.

We could try to migrate the data into a Directus instance or we could create a middleware between the two databases to convert the data.
In both cases we could have a wizard to guide user.

**Reply by rijkvanzanten** [reply]
> But, things like, file associations, translations, associate proper interfaces, etc. Those are all tasks everyone need to do in order to use an existing database.

Yes on the file associations, but you don't _need_ to do any of the other stuff in order to use an existing database with Directus.

**Reply by hugopvl** [reply]
But in this case I didn't want to use the existing DB, I wanted to migrate everything to a new one, and it's already done :)

in the case of files associated with items, I decided as follows:

As all the files contained the item ID in the name, I uploaded all of them through directus files: 15,000 files and 400GB, I just want to emphasize that it went beautifully, if you needed a test drive it went beautifully :) I uploaded in blocks 1000, some blocks with 70GB and it's super fast.

After that I updated the tables I needed, and it was easy to put the files in the items ;)
